### ISSUE TITLE
Compound Indexes

### ISSUE THREAD

**Issue by rmanibus** [seald/nedb#26]
**Is your feature request related to a problem? Please describe.**
Today it is only possible to create an index on one single field, but it is often required to index on multiple fields.

**Describe the solution you'd like**

I implemented compound indexes a while ago on my own nedb fork. (https://github.com/rmanibus/nedb/pull/1)

this is how it works:

```
db.ensureIndex({ fieldName: ["field1", "field2"] }, function (err) {});
```

my branch would need to be rebased but I think this can easily be integrated in that project.

**Comment by rmanibus** [comment]
I also have a github workflow ready for this project to execute tests on PR if you want

**Comment by tex0l** [comment]
This has been released in [`v4.0.0`](https://github.com/seald/nedb/commit/eff2c4a91e185be3f068667d024fe0fbd57fe870).
